src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -109,6 +110,30 @@ private protected PredictionTransformerBase(IHost host, ModelLoadContext ctx)
             TrainSchema = loader.Schema;
         }
 
+        [BestFriend]
+        private protected PredictionTransformerBase(IHost host, ModelLoadContext ctx, TModel model)
+        {
+            Host = host;
+
+            // *** Binary format ***
+            // model: prediction model.
+            // stream: empty data view that contains train schema.
+            // id of string: feature column.
+
+            Model = model;
+
+            // Clone the stream with the schema into memory.
+            var ms = new MemoryStream();
```

Comment:
Each new constructor here seems to have a decent amount of duplicated code with the existing constructor. Can it be refactored to share initialize logic?

Reply:
I think it can be done. I just didn't want to modify the existing constructors. But I will change this.

Reply:
In my latest update I've created initialize logic methods, except for the following classes:
- `RegressionPredictionTransformer<>` and `RankingPredictionTransformer<>`, because their constructors were already one liners, and didn't see much use in refactoring them. If requested I can change them as well.
- `SingleFeaturePredictionTransformerBase<>`, because in its constructors there are several assignments to readonly properties, and I believe it wouldn't be possible to assign to them outside of the constructors (in an initialize logic method) without making other changes to the properties themselves. So I am unsure if should also go on and refactor this class.

Reply:
In my latest update I've created initialize logic methods, except for the following classes:
-` RegressionPredictionTransformer<>` and `RankingPredictionTransformer<>`, because their constructors were already one liners, and didn't see much use in refactoring them. If requested I can change them as well.
- `SingleFeaturePredictionTransformerBase<>`, because in its constructors there are several assignments to readonly properties, and I believe it wouldn't be possible to assign to them outside of the constructors (e.g. in an initialize logic method) without making other changes to the properties themselves. So I am unsure if should also go on and refactor this class.

Reply:
I think the current refactoring is enough. Thanks for cleaning it up a bit.